hilti/src/rt/types/address.cc

```diff
@@ -87,12 +93,18 @@ std::variant<struct in_addr, struct in6_addr> Address::asInAddr() const {
     cannot_be_reached();
 }
 
-bool Address::operator==(const Address& other) const { return _a1 == other._a1 && _a2 == other._a2; }
+bool Address::operator==(const Address& other) const {
+    // NOTE: `_family` is not checked here as IPv4 and IPv6 addresses can be equivalent.
```

Comment:
I can see doing this either way, so ok.

Reply:
In the current implementation addresses still have a slight preference towards IPv4 since e.g., `Adress("::")` constructs an address `0.0.0.0`. Not including `_family` here makes it slightly less easier for us to compare values, but it shouldn't be hard to change the behavior here (also why I included the `NOTE`).